src/operations/query/r4.js

```diff
@@ -123,23 +123,45 @@ class R4SearchQueryCreator {
                             field: propertyObj.field
                         }
                     );
-                    /**
-                     * @type {string | string[]}
-                     */
-                    let queryParameterValue = args[`${queryParameter}`];
-                    queryParameterValue = convertGraphQLParameters(
-                        queryParameterValue,
-                        args,
-                        queryParameter
-                    );
+
                     // if just a query parameter is passed then check it
-                    if (queryParameterValue) {
+                    let actualArg = args[queryParameter] ? queryParameter : null;
+                    if (!actualArg) {
+                        for (const argName in args) {
+                            if (argName.startsWith(`${queryParameter}:`)) {
+                                actualArg = argName;
+                                break;
+                            }
+                        }
+                    }
+
+                    if (actualArg) {
+                        /**
+                         * @type {string | string[]}
+                         */
+                        let queryParameterValue = args[actualArg];
+                        queryParameterValue = convertGraphQLParameters(
+                            queryParameterValue,
+                            args,
+                            queryParameter
+                        );
+
                         // handle id differently since it is a token, but we want to do exact match
                         if (queryParameter === '_id') {
                             filterById({
-                                queryParameterValue, and_segments, propertyObj, columns
+                                queryParameterValue, and_segments: total_and_segments, propertyObj, columns: totalColumns
+                            });
+                            continue; // skip processing rest of this loop
+                        } else if (args[`${queryParameter}:text`]) {
+                            // handle partial text search differently because I can't figure out how to get it to combine with other search types
+                            filterByPartialText({
```

Comment:
I wrestled with the partial text search for a while. I can't get it to work without modifying the final result in the columns Set. In the end, I think it's safer to keep the partial text search working as today. So I moved it up to this point in the file.